src/chocolatey/infrastructure.app/configuration/ChocolateyConfiguration.cs

```
@@ -58,6 +61,86 @@ public ChocolateyConfiguration()
+            }
+
+            var t = this.GetType();
+
+            foreach (var property in t.GetProperties(BindingFlags.Public | BindingFlags.Instance))
+            {
+                try
+                {
+                    var originalValue = property.GetValue(_originalConfiguration, new object[0]);
+
+                    if (removeBackup || property.DeclaringType.IsPrimitive || property.DeclaringType.IsValueType || property.DeclaringType == typeof(string))
+                    {
+                        // If the property is a primitive, a value type or a string, then a copy of the value
+                        // will be created by the .NET Runtime automatically, and we do not need to create a deep clone of the value.
+                        // Additionally, if we will be removing the backup there is no need to create a deep copy
+                        // for any reference types, as such we also sets the reference itself so it is not needed
+                        // to allocate more memory.
+                        property.SetValue(this, originalValue, new object[0]);
+                    }
+                    else if (originalValue != null)
+                    {
+                        // We need to do a deep copy of the value so it won't copy the reference itself,
+                        // but rather the actual values we are interested in.
+                        property.SetValue(this, originalValue.deep_copy(), new object[0]);
+                    }
+                    else
+                    {
+                        property.SetValue(this, null, new object[0]);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    this.Log().Debug("Unable to restore the value for property '{0}' with the message '{1}'", property.Name, ex.Message);

```

Is this possibly a scenario where a more present message should be presented?

I guess the overarching question I have is: what are the possible consequences if a value is unable to be restored?

---

> Is this possibly a scenario where a more present message should be presented?
> 
> I guess the overarching question I have is: what are the possible consequences if a value is unable to be restored?

The consequence if the value can not be restored would be that it would still contain the value that was changed.
However, I was not able to produce a case where this would fail in the current codebase, as such I only added a debug logging for this in the rare case it could fail.

---

I have changed this to rethrow the exception as an application exception instead of silently ignoring it.